### PR TITLE
Stops empty ToC block appearing when no headings to link to are found.

### LINKED ISSUE #37: On this page title displays even if no links
image

Ideally we should hide the TOC block if there are no H2s

### PR DESCRIPTION
**Pull request by rupertj**
Fixes #37.

### CODE REVIEW

File: config/optional/block.block.localgov_publications_toc_block.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ dependencies:
     - localgov_publications
     - node
   theme:
-    - localgov_scarfolk
+    - localgov_base
 id: localgov_publications_toc_block
-theme: localgov_scarfolk
+theme: localgov_base
```
Comment (finnlewis) [inline config/optional/block.block.localgov_publications_toc_block.yml]: This looks to place the block for localgov_base but not for localgov_scarfolk. 

I think we want to default to localgov_scarfolk, but you can include a number of optional configs for each desired theme I think.

File: tests/src/Functional/TocBlockTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * Functional tests for the TocBlock.
+ */
+class TocBlockTest extends BrowserTestBase {
+
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'localgov_base';
```
Comment (finnlewis) [inline tests/src/Functional/TocBlockTest.php:19]: Maybe test against localgov_scarfolk too? Or include block placement for both localgov_scarfolk and localgov_base?

**Review by finnlewis — CHANGES_REQUESTED** [review]
Also, we've not got the default github workflows tests running yet, so worth running phpcs locally.

Just did and there are a few coding standards things to fix: 

```
finn@Oobuntoo:~/sites/l.localgov/MY_PROJECT$ lando phpcs web/modules/contrib/localgov_publications/
............................................EE...W.E.. 54 / 54 (100%)



FILE: ...les/contrib/localgov_publications/localgov_publications.info.yml
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 14 | ERROR | [x] Expected 1 newline at end of file; 2 found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------


FILE: ...trib/localgov_publications/tests/src/Functional/TocBlockTest.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 33 | ERROR | [x] Missing function doc comment
 48 | ERROR | [x] There must be exactly one blank line before the
    |       |     tags in a doc comment
 74 | ERROR | [x] Expected 1 blank line after function; 0 found
 75 | ERROR | [x] The closing brace for the class must have an empty
    |       |     line before it
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------


FILE: ...dules/contrib/localgov_publications/localgov_publications.module
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 44 | WARNING | Line exceeds 80 characters; contains 82 characters
----------------------------------------------------------------------


FILE: ...ules/contrib/localgov_publications/localgov_publications.install
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 1 | ERROR | [x] Missing file doc comment
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 801ms; Memory: 12MB
```

**Comment by finnlewis** [comment]
The block visibility does work though, so thanks, for this! :smiley_cat:

File: config/optional/block.block.localgov_publications_toc_block.yml (hunk 1 above)
Comment (rupertj) [reply config/optional/block.block.localgov_publications_toc_block.yml]: I've added one for each theme now.

File: tests/src/Functional/TocBlockTest.php (hunk 2 above)
Comment (rupertj) [reply tests/src/Functional/TocBlockTest.php:19]: I've kept the test as is, and there's a block config for both themes now.

**Comment by rupertj** [comment]
I've merged 1.x in here with the proper CI pipeline setup, so we should see the tests pass in a bit...

**Review by finnlewis — APPROVED** [review]
Looks good to me, thanks @rupertj :smile:

**Pull request merged**